### PR TITLE
Fix error message modelchain.infer_temperature_model error to be more informative

### LINKED ISSUE #1946: Fix error message in ModelChain pertaining to temp_model inference
**Describe the bug**
Error message is incorrect.
The message is:

`ValueError: could not infer temperature model from system.temperature_model_parameters. Check that all Arrays in system.arrays have parameters for the same temperature model. Common temperature model parameters: set().`

However, in code below we are creating a PVSystem without using `Arrays`.  So the second sentence in the message above is not describing the proper way to fix this issue.

**To Reproduce**
```python

from pvlib import pvsystem
from pvlib.location import Location
from pvlib.modelchain import ModelChain

location = Location(latitude=32.2, longitude=-110.9)

modules = pvsystem.retrieve_sam('cecmod')
module_parameters = modules['Canadian_Solar_Inc__CS5P_220M']
inverter_parameters = {'pdc0': 5000, 'eta_inv_nom': 0.96}
system = pvsystem.PVSystem(inverter_parameters=inverter_parameters,
                           module_parameters=module_parameters)

# create the model chain
modelchain = ModelChain(
    system, location, name='dummy_name', aoi_model="physical", temperature_model="pvsyst"
)
```

**Expected behavior**
Error message correctly explains what needs to be changed in `pvsystem` insantiation in order for `ModelChain` to run.

**Versions:**
 - ``pvlib.__version__``: 0.10.2 
 - python: 3.9.6

**Additional context**
This error message also appears in other contexts.  Namely for other model types:
AC
AOI
DC
spectral

So the error messages for those models should be fixed as well.  Search for 'could not infer' to see which messages.

### PR DESCRIPTION
**Pull request by matsuobasho**
- [x] Closes #1946
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: pvlib/modelchain.py [hunk 1]
```diff
@@ -1128,15 +1128,15 @@ def infer_temperature_model(self):
...
-            raise ValueError(f'could not infer temperature model from '
-                             f'ModelChain.system.  '
-                             f'If Arrays are used to construct the PVSystem, '
-                             f'check that all Arrays in '
-                             f'ModelChain.system.arrays '
-                             f'have parameers for the same temperature model. '
-                             f'If Arrays are not used, check that the PVSystem '
-                             f'attributes `racking_model` and `module_type` '
-                             f'are valid.')
+            raise ValueError('could not infer temperature model from '
```
Comment (cwhanse) [inline pvlib/modelchain.py:1131]:
```suggestion
            raise ValueError('Could not infer temperature model from '
```

**Review by cwhanse — COMMENTED** [review]
Adjust the match text [here ](https://github.com/pvlib/pvlib-python/blob/62de6547c73f8b254f924f303ae2255b4b6540a2/pvlib/tests/test_modelchain.py#L1318)and I think this is done.

File: pvlib/modelchain.py (hunk 1 above)
Comment (kandersolar) [reply pvlib/modelchain.py]: @matsuobasho this capitalization change will also require the tests to be updated.  See the test failures here: https://github.com/pvlib/pvlib-python/actions/runs/8146219415/job/22264244062?pr=1977#step:9:134

File: pvlib/modelchain.py [hunk 2]
```diff
@@ -1128,11 +1128,15 @@ def infer_temperature_model(self):
...
-                             f'system.temperature_model_parameters. Check '
-                             f'that all Arrays in system.arrays have '
-                             f'parameters for the same temperature model. '
-                             f'Common temperature model parameters: {params}.')
+            raise ValueError('Could not infer temperature model from '
+                             'ModelChain.system.  '
+                             'If Arrays are used to construct the PVSystem, '
+                             'check that all Arrays in '
+                             'ModelChain.system.arrays '
+                             'have parameers for the same temperature model. '
```
Comment (kandersolar) [inline pvlib/modelchain.py:1136]:
```suggestion
                             'have parameters for the same temperature model. '
```

**Comment by kandersolar** [comment]
@matsuobasho the tests on this PR are failing due to the match text needing to be updated as mentioned above.  After that I think this PR will be ready to merge.

**Comment by matsuobasho** [comment]
@kandersolar thank you for the guidance on this, your message about the capitalization in the tests earlier in the week slipped past me.  Will address.

**Review by kandersolar — APPROVED** [review]
LGTM, thanks @matsuobasho!

**Pull request merged**